### DISCUSSION TITLE
A few questions before jumping in

### DISCUSSION THREAD

**Discussion by tangoTH** [immich-app/immich#7770] [General]
Not sure where to ask - Tried on reddit (no answer) so trying here

I have a NAS drive with a directory structure with Home, Work, Other at the root and then subdirectories. Meaning i've pre-sorted everything into folders.

    Can I import this structure into Immich and keep the existing structure or will Immich just import and store everything in it's database? In the past, i've used Jriver and it was like this - maintaining the structure

    If a family member uploads photos to Immich, how do I then get those photos out and synced with the structure store mentioned earlier? EG. How do I backup the entire photo gallery as the JPG files that they are. Thinking for 5-10 years later, I'd like them preserved as original in their folders, not tied to a software.

Many thanks for any insights

Edit: Looks like a External Library may be possible. Need to read how it works. Anyone have an example?
https://immich.app/docs/features/libraries/

Also is there really no way to upload a single image or a group of images from an Android phone to create a new album? This is surprising and a basic use case IMO

**Comment by alextran1502** [comment]
For your first question, the answer is yes, and external library is exactly what you are looking for.

I am not quite understanding your second question. You can upload a single photo, or group of photos, or the whole album from the phone

**Reply by tangoTH** [reply]
Thank you for the response.  I appreciate it.

I'll try to be a little clear.

1) 
I have my external NAS with a structured library set up.  Can Immich import those folders (Home/Family, Home/Projects,  Home/Trips/Japan2022, Home/Trips/Europe2023).   and keep those same groups or tags?  Or will they all be imported as a big set of pictures and i'll have no way to group them?  

2) Will Immich copy all of the copy photo's into its own database and keep the originals there? What if I add more photos to the above? Will it import those too?

3) If I upload a photo into Immich, where is it stored? If it is an internal database, how do I dump the database and get all the photos as original quality?   While I'd like to use Immich as a gallery, I also need to save the files that family members upload


4) On Android, how to upload a single photo to a specific group.  Ie picture of a my dog to the Dog folder/Tag?

Thank you

**Reply by mertalev** [reply]
> I have my external NAS with a structured library set up. Can Immich import those folders (Home/Family, Home/Projects, Home/Trips/Japan2022, Home/Trips/Europe2023). and keep those same groups or tags? Or will they all be imported as a big set of pictures and i'll have no way to group them?

It won't group them unfortunately. We do want to expand on albums and tagging in the coming months, though, so this is an area of improvement.

> Will Immich copy all of the copy photo's into its own database and keep the originals there? What if I add more photos to the above? Will it import those too?

When using external libraries, the original asset isn't copied over. Thumbnails, transcodes, etc. will be created in Immich's upload location (not in the external library's folder). We also have a watching feature for libraries to automatically sync with file changes.

> If I upload a photo into Immich, where is it stored? If it is an internal database, how do I dump the database and get all the photos as original quality? While I'd like to use Immich as a gallery, I also need to save the files that family members upload

They go to your upload location, the folder you specify where uploads should go. They're regular files and not in any arcane format, so you can easily copy and use them outside of Immich. If storage template mode is enabled, they'll also be organized into year, album name, etc. as you like.

> On Android, how to upload a single photo to a specific group. Ie picture of a my dog to the Dog folder/Tag?

You just add that image to your Dog album in the app and it'll sync with the server.

**Comment by bossanova808** [comment]
It's more yes _and_ no, really, isn't it?  

I.e. When using an external library, there is as yet no way for Immich to automatically to create albums from the existing on  disk folder structure (or even to then view images by folder structure)

(Sorry to butt in/contradict but this is exactly what I have been trying out today...)

As of this moment, there are two popular discussions that are relevant:

https://github.com/immich-app/immich/discussions/4279 (creation of albums by folder structure for external libraries)
And
https://github.com/immich-app/immich/discussions/5409 (view by folder although for me, the former is a lot more interesting)

(There are some scripts in those discussions to help with the task, but I would want it to go further - if I re-arrange the albums on disk etc, I'd want that picked up by Immich etc.)

**Reply by tangoTH** [reply]
Jriver does this and it's what i'm used to.  Makes manual tagging soooo much easier.  ALso automated tagging based on folder

ie
Home
         / Dog
         / House Construction
        / Kids
        

Can someone advise on the other questions? I'm looking to confirm functionality before i Jump in
